### PR TITLE
Export defaults from PooledByteBufAllocator static fields - tuning pa…

### PR DESCRIPTION
**Pull request by slachiewicz**
…rameters for org.apache.spark.network.util.NettyUtils Now, Spark must modify access to private fields.

### CODE REVIEW

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 1]
```diff
@@ -280,6 +280,28 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+    }
+    public static int getDefaultTinyCacheSize() {
+        return DEFAULT_TINY_CACHE_SIZE;
+    }
+    public static int getDefaultSmallCacheSize() {
+        return DEFAULT_SMALL_CACHE_SIZE;
+    }
+    public static int getDefaultNormalCacheSize() {
+        return DEFAULT_NORMAL_CACHE_SIZE;
+    }
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: we not use get* prefixes in netty. Please remove these. Also please add javadocs.

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 2]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
 
         return toLeakAwareBuffer(buf);
     }
+    /**
+     *  Default number of heap areanas - System Property: io.netty.allocator.numHeapArenas - default 2 * cores
+     */
+    public static int defaultNumHeapArena() {
```
Comment (Scottmitch) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: Should we make this part of the internal API somehow @normanmaurer ?

Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: what you mean with "the internal API" ?

Comment (Scottmitch) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: I see now that we are exposing other internal details of this class (`smallCacheSize`, etc...) so nvm.

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 3]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+        return DEFAULT_TINY_CACHE_SIZE;
+    }
+    /**
+     *  Default small cache size - System Property: io.netty.allocator.smallCacheSize - default 256
+     */
+    public static int defaultSmallCacheSize() {
+        return DEFAULT_SMALL_CACHE_SIZE;
+    }
+    /**
+     *  Default normal cache size - System Property: io.netty.allocator.normalCacheSize - default 64
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 4]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+        return DEFAULT_MAX_ORDER;
+    }
+    /**
+     *  Default tiny cache size - System Property: io.netty.allocator.tinyCacheSize - default 512
+     */
+    public static int defaultTinyCacheSize() {
+        return DEFAULT_TINY_CACHE_SIZE;
+    }
+    /**
+     *  Default small cache size - System Property: io.netty.allocator.smallCacheSize - default 256
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 5]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+        return DEFAULT_PAGE_SIZE;
+    }
+    /**
+     *  Default maximum order - System Property: io.netty.allocator.maxOrder - default 11
+     */
+    public static int defaultMaxOrder() {
+        return DEFAULT_MAX_ORDER;
+    }
+    /**
+     *  Default tiny cache size - System Property: io.netty.allocator.tinyCacheSize - default 512
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 6]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+        return DEFAULT_NUM_DIRECT_ARENA;
+    }
+    /**
+     * Default buffer page size - System Property: io.netty.allocator.pageSize - default 8192
+     */
+    public static int defaultPageSize() {
+        return DEFAULT_PAGE_SIZE;
+    }
+    /**
+     *  Default maximum order - System Property: io.netty.allocator.maxOrder - default 11
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 7]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
+        return DEFAULT_NUM_HEAP_ARENA;
+    }
+    /**
+     *  Default numer of direct arenas - System Property: io.netty.allocator.numDirectArenas - default 2 * cores
+     */
+    public static int defaultNumDirectArena() {
+        return DEFAULT_NUM_DIRECT_ARENA;
+    }
+    /**
+     * Default buffer page size - System Property: io.netty.allocator.pageSize - default 8192
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 8]
```diff
@@ -279,6 +279,48 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
...
         return toLeakAwareBuffer(buf);
     }
+    /**
+     *  Default number of heap areanas - System Property: io.netty.allocator.numHeapArenas - default 2 * cores
+     */
+    public static int defaultNumHeapArena() {
+        return DEFAULT_NUM_HEAP_ARENA;
+    }
+    /**
+     *  Default numer of direct arenas - System Property: io.netty.allocator.numDirectArenas - default 2 * cores
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: missing `.` on EOL

**Comment by normanmaurer** [comment]
@slachiewicz can you also squash the commits into one and sign the ICLA:
http://netty.io/s/icla 

And please update the commit message to follow our template:
http://netty.io/wiki/writing-a-commit-message.html

**Comment by normanmaurer** [comment]
@slachiewicz ping ;)

**Comment by slachiewicz** [comment]
Done, with lots of pain with git ;-)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (ad008f6831b1d233d9669a99fcec74ccf44a9c72), 4.1(a18416df60809d5ec9c703bfbc15658a8d4443fc) and master (a6816bd59ef2923d82af5b7a7f3b722507a56a3a)

**Comment by normanmaurer** [comment]
@slachiewicz thanks!

**Pull request closed without merging**